custom_md_loader/classes/MetadataApexApiLoader.cls

```diff
@@ -0,0 +1,230 @@
+/* 
+ * Copyright (c) 2016, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license. 
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+ 
+public with sharing class MetadataApexApiLoader extends MetadataLoader {
+
+	public MetadataDeployStatus mdDeployStatus {get;set;}
+	public MetadataDeployCallback callback {get;set;}
+	
+	public static Id jobId1 {get;set;}
+	public static Metadata.DeployStatus deployStatus1 {get;set;}
+	public static boolean success1 {get;set;}
+
+    public MetadataApexApiLoader() {
+    	this.mdDeployStatus = new MetadataApexApiLoader.MetadataDeployStatus();
+    	this.callback = new MetadataDeployCallback(); 
+    } 
+    
+    public MetadataApexApiLoader.MetadataDeployStatus getMdDeployStatus() {
+    	return this.mdDeployStatus;
+	}
+	
+	public MetadataApexApiLoader.MetadataDeployCallback getCallback() {
+    	return this.callback;
+	}
+    
+    public override void migrateAsIsWithObjCreation(String csName, String cmtName) {
+		List<MetadataResponse.Message> messages = new List<MetadataResponse.Message>();
+		messages.add(new MetadataResponse.Message(100, 'Not Supported!!!'));
+		response.setIsSuccess(false);
+	    response.setMessages(messages);  
+    }
+    
+    public override void migrateAsIsMapping(String csName, String cmtName) {
+    	super.migrateAsIsMapping(csName, cmtName);
+    	buildResponse();
+    }
+    
+    public override void migrateSimpleMapping(String csNameWithField, String cmtNameWithField) {
+    	super.migrateSimpleMapping(csNameWithField, cmtNameWithField);
+    	buildResponse();
+    }
+    
+    public override void migrateCustomMapping(String csName, String cmtName, String mapping) {
+    	super.migrateCustomMapping(csName, cmtName, mapping);
+    	buildResponse();
+    }
+    
+    private void buildResponse() {
+    	if(response.IsSuccess()) {

```

Comment:
Do you need an else case for when response is error?

Reply:
else not needed, since response would have been build with failure by now that we use to display in UI.